### PR TITLE
fix: use an `a` tag when `RouterLink` links to different browsing context

### PR DESCRIPTION
**Pull request by starsirius**
The type of this PR is: Fix

### Description

Integrity [reported](https://app.circleci.com/pipelines/github/artsy/integrity/23145/workflows/b3acd151-83b8-42b5-ad2d-07d442012cbc/jobs/114822) that the condition of sales link in the auction registration modal does not open in a new tab. Upon [investigation](https://artsy.slack.com/archives/C02BC3HEJ/p1714494569053839), I think the [recent `found` upgrade](https://github.com/artsy/force/pull/13817) introduced a breaking change that the [`target` prop](https://github.com/artsy/force/blob/f53966c99ee0c55a15e9a850399cb2dec275f60a/src/Apps/Auction/Components/Form/ConditionsOfSaleCheckbox.tsx#L39) of a `RouterLink` is dropped when rendering the link.

I think [this commit](https://github.com/4Catalyzer/found/commit/399ef8e28d438e3e426f222ae983851f42cf4b2a#) in `found` introduced the change, in which [`target` was taken out of the `props`](https://github.com/4Catalyzer/found/commit/399ef8e28d438e3e426f222ae983851f42cf4b2a#diff-08114c649a298b4f09c94e7dd569d18b93d3dbb14b8bf12061093a167052379aR17) and [was not passed down to the underlying component](https://github.com/4Catalyzer/found/commit/399ef8e28d438e3e426f222ae983851f42cf4b2a#diff-08114c649a298b4f09c94e7dd569d18b93d3dbb14b8bf12061093a167052379aR101).

My attempt to reproduce and fix the breaking change is [here](https://github.com/starsirius/found/commit/9c6b345d526efada34677a65e943d1e711e1692c). I'll submit a PR to the repo and get feedback.

~Before it gets merged/released, let's patch it for now.~ Based on the [feedback](https://github.com/artsy/force/pull/13829#issuecomment-2088317364), instead of patching `found`, I updated the `RouterLink` to be smarter when choosing to render a `Link` or an `a` tag directly in [17b9772](https://github.com/artsy/force/pull/13829/commits/17b9772eaa03e42da429580d6552b9b2809921f9).

Screenshot 2024-04-30 at 9 13 33 PM

### CODE REVIEW

**Comment by mzikherman** [comment]
Maybe something to do with the way `found` is being packaged? For instance, the existing `relay-mock-network-layer` patch modifies `node_modules/relay-mock-network-layer/index.js` - seems sort of vanilla, whereas this patch targets the `cjs` and `esm` sub-directories. Do both need to be patched or maybe there's a better `index.js`-type file that can be patched?

**Review by mzikherman — DISMISSED** [review]
Great writeup and investigation - what you've found and this fix makes sense to me! Nice work. We should totally open a PR upstream, but in the meantime a patch (or even fork) totally works.

**Comment by dzucconi** [comment]
The whole point of the router link is that it interfaces with the router. A target="_blank" means it can just be a regular `a` tag. So this isn't needed. And the whole patch thing is a nightmare to deal with when upgrading stuff.

**Comment by dzucconi** [comment]
(Admittedly, it is annoying to have to think about)

**Comment by starsirius** [comment]
> The whole point of the router link is that it interfaces with the router. A target="_blank" means it can just be a regular a tag. So this isn't needed. And the whole patch thing is a nightmare to deal with when upgrading stuff.

@dzucconi Interesting! So maybe the `RouterLink` can be smarter [when choosing to render a `Link` from `found` or an `a` directly](https://github.com/artsy/force/blob/46b85204de2972310b2c8fae4e54dea3a37a267d/src/System/Router/RouterLink.tsx#L38C9-L43). Let me give that a try!

**Comment by mzikherman** [comment]
Oh yea @dzucconi in hindsight that makes a lot of sense. If `target="_blank"` it can be a `RouterUnawareLink`.

**Comment by starsirius** [comment]
@dzucconi @mzikherman Alright, pushed up the second attempt to _not_ depend on the upstream breaking change! See if this makes sense!

**Review by dzucconi — APPROVED** [review]

File: src/System/Router/__tests__/RouterLink.jest.tsx [hunk 1]
```diff
@@ -34,16 +35,59 @@ describe("RouterLink", () => {
         ]}
       />
     )
-  }
+    await flushPromiseQueue()
 
-  // TODO: uncomment this not sure why its failing
-  it.skip("uses the <Link> component if within a router context", async () => {
```
Comment (starsirius) [inline src/System/Router/__tests__/RouterLink.jest.tsx:40]: Happy to re-enable this! 😎

**Comment by starsirius** [comment]
OK Let's give it a try! 💪

File: src/System/Router/__tests__/RouterLink.jest.tsx (hunk 1 above)
Comment (mzikherman) [reply src/System/Router/__tests__/RouterLink.jest.tsx:40]: Awesome!

**Review by damassi — COMMENTED** [review]
Thanks for the fix @starsirius

**Pull request merged**